worker/src/handlers/aggregationProcessor/aggregationProcessor.js

```diff
@@ -214,12 +237,34 @@ const getAddPipeline = async ({
     };
   }
 
+  // MIDDLE
+
+  let addToMiddle;
+  if (model.lastRun) {
+    addToMiddle = {
+      $and: [
+        {
+          timestamp: {
+            $gte: moment(model.fromTimestamp || now).toDate(),
+            $lt: getAddFromTimestamp({ model, now }).toDate()
+          }
+        },
+        {
+          stored: {
+            $gte: moment(model.lastRun).toDate(),
+          }
+        }
+      ]
+    };
+  }
+
   return [
     {
       $match: {
         $or: [
           { timestamp: addToFrontPipeline },
-          { timestamp: addToEnd }
+          { timestamp: addToEnd },
+          ...(addToMiddle ? [addToMiddle] : []),
```

Comment:
See earlier comment about moving the ternary earlier.
```suggestion
          ...addToMiddle,
```